src/user/controller/user.controller.ts

```diff
@@ -1,36 +1,62 @@
 import {
+  Logger,
   BadRequestException,
   Body,
   Controller,
   HttpStatus,
   Inject,
   NotFoundException,
   Post,
+  Get,
   Res,
+  Param,
+  ImATeapotException,
 } from '@nestjs/common';
 import { ApiTags } from '@nestjs/swagger';
 import { User } from '../schema/user.document';
 import { IUserService } from '../service/user.service.interface';
+import { error } from 'console';
 
 @ApiTags('users')
 @Controller('users')
 export class UserController {
+  private readonly logger = new Logger(UserController.name);
   constructor(
     @Inject(IUserService) private readonly userService: IUserService,
   ) {}
   @Post()
   async save(@Res() response: any, @Body() user: User): Promise<User> {
-    try {
+    try { 
       const newUser = await this.userService.save(user);
+      
       return response.status(HttpStatus.CREATED).json(newUser);
     } catch (e: any) {
       if (e instanceof NotFoundException) {
-        return response.status(HttpStatus.NOT_FOUND).json();
+        return response.status(HttpStatus.NOT_FOUND).json({error: 'Email já cadastrado.'});
```

Comment:
O correto aqui seria retornar 400 (Bad Request) se estiver tentando criar uma conta usando um e-mail já cadastrado. Not Found é um status code para entidades esperadas que existam, porém que não foram encontradas

Reply:
É verdade hehe, o Not Found não faz muito sentido nesse contexto, eu tinha procurado um erro que ficasse bom nesse caso, mas eu não tinha pensado em nenhum hehe.